### PR TITLE
turn off asserts for rails api mode

### PR DESCRIPTION
**Pull request by pandamako**
I would like to app pghero to rails5 api app and use from console. And got error with assets.

### CODE REVIEW

**Comment by ankane** [comment]
Hey @pandamako, thanks for the PR 👍 

Just to make sure I understand, you want to use PgHero with Rails but without the web interface?

Also, I'm not sure previous versions of Rails will like `app.config.api_only`.

**Comment by pandamako** [comment]
Yes, you're right. I'm going to use PgHero with Rails without the web interface, just `json` responses.

File: lib/pghero/engine.rb [hunk 1]
```diff
@@ -3,13 +3,15 @@ class Engine < ::Rails::Engine
...
 
     initializer "pghero", group: :all do |app|
-      if defined?(Sprockets) && Sprockets::VERSION >= "4"
-        app.config.assets.precompile << "pghero/application.js"
-        app.config.assets.precompile << "pghero/application.css"
-      else
-        # use a proc instead of a string
-        app.config.assets.precompile << proc { |path| path == "pghero/application.js" }
-        app.config.assets.precompile << proc { |path| path == "pghero/application.css" }
+      unless app.config.try? :api_only
```
Comment (ankane) [inline lib/pghero/engine.rb]: Think this should be `try` without the question mark.

**Comment by ankane** [comment]
Thanks @pandamako, looks good. Should have asked at the start, but can you paste the error you're seeing without the changes in PR? I'd like to see if it's impossible to use the web interface in API mode, or it's something specific to your app. If it is possible in API mode, this change would break existing apps.

**Comment by ankane** [comment]
Hey @pandamako, I decided to go with a safer approach. Thanks for all your work on this.

**Pull request closed without merging**